src/Datadog.Trace/Logging/LibLogScopeEventSubscriber.cs

```diff
@@ -271,13 +271,19 @@ private static void InitResolvers()
                     CustomLog4NetLogProvider.IsLoggerAvailable,
                     () => new CustomLog4NetLogProvider()));
 
-            // Register the custom NLog provider
+            // Register the custom NLog providers
             LogProvider.LogProviderResolvers.Insert(
                 0,
                 Tuple.Create<LogProvider.IsLoggerAvailable, LogProvider.CreateLogProvider>(
                     CustomNLogLogProvider.IsLoggerAvailable,
                     () => new CustomNLogLogProvider()));
 
+            LogProvider.LogProviderResolvers.Insert(
+                1,
+                Tuple.Create<LogProvider.IsLoggerAvailable, LogProvider.CreateLogProvider>(
+                    FallbackNLogLogProvider.IsLoggerAvailable,
+                    () => new FallbackNLogLogProvider()));
+
```

Comment:
It's a minor thing, but I find it a bit confusing to be using index `1` here when we use `0` for all the other providers (meaning that they are added in increasing precedence order). The fallback provider is the only one that breaks that pattern.

That said, I totally understand why you went for that approach, as the fact we have to do Insert 0 is a bit annoying!

Reply:
I second this. Since the order of the providers is important, we need to make this code consistent. I don't mind if we use `Insert(0)` or `Insert(0,1,2,3,...)`, but let's use the same for all providers

Reply:
I went with the `Insert(0,1,2,3,...)` approach in my latest commit. Let me know if that looks good to you